src/Functions/FunctionsHashing.h

```diff
@@ -1048,11 +1081,28 @@ class FunctionAnyHash : public IFunction
         {
             auto value = col_from_const->template getValue<FromType>();
             ToType hash;
-            if constexpr (std::is_same_v<ToType, UInt64>)
-                hash = IntHash64Impl::apply(bit_cast<UInt64>(value));
+            if constexpr (std::endian::native == std::endian::little)
```

Comment:
Could you please apply the same structural changes from above also to the code here? Basically try to hide the LE/BE distinction as deeply in the nested ifs as possible.

Reply:
Done.